types/testing-library__jest-dom/index.d.ts

```diff
@@ -0,0 +1,75 @@
+// Type definitions for @testing-library/jest-dom 4.0
+// Project: https://github.com/testing-library/jest-dom
+// Definitions by: Ernesto García <https://github.com/gnapse>
+//                 John Gozde <https://github.com/jgoz>
+//                 Seth Macpherson <https://github.com/smacpherson64>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+/// <reference types="jest" />
+
+// CONTRIBUTOR NOTE: When adding/updating matchers, ensure you handle
+// them in both the Global section and the Module section
+
+export type ValidElement = HTMLElement | SVGElement;
+
+// Global types (for `import "@testing-library/jest-dom/extend-expect"`)
+
+declare global {
+    namespace jest {
```

Comment:
Move this global declaration change into file `extend-expect.d.ts` since the wayt to handle is using `import "@testing-library/jest-dom/extend-expect"`
Refer to https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#library-file-layout

Reply:
Will that make the global declarations available to user code even if the `import` statement does not appear in a TypeScript-compiled file?

Typically, this plugin is configured by including that import in a JS file that is only used by Jest (and Jest does not support reading TS files directly for configuration). The purpose of moving these declarations to DT is so that the global type definitions are just "available" without forcing users to add a separate `.d.ts` file that includes the import statement.